### PR TITLE
fix(dashmate): cleanup zerossl certs command

### PR DESCRIPTION
**Pull request by shumkov**
## Issue being fixed or feature implemented

Due to recent bugs (#2297 and #2296), the dashmate helper created thousands of draft certificates. Free accounts allow only 3 certificates and paid 10000+. Multiple users reported that these limits were reached. We need to provide a simple way to cleanup accounts from created certs.

## What was done?

- Implemented the `dashmate ssl cleanup` command

## How Has This Been Tested?

Running command with account contained 10123 drafts.

## Breaking Changes

None

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [x] I have made corresponding changes to the documentation if needed

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

## Release Notes

- **New Features**
  - Introduced a `CleanupCommand` for managing Zero SSL certificate cleanup, allowing users to cancel drafted or pending validation certificates.
  - Added a factory function for creating cleanup tasks related to ZeroSSL certificates, improving task management and execution.

- **Improvements**
  - Enhanced the `listCertificates` function to support pagination, enabling users to retrieve results across multiple pages.
  - Updated the `ObtainCommand` to streamline task execution by changing the task to synchronous.
  - Modified the `scheduleRenewZeroSslCertificate` function to improve control flow by removing asynchronous handling.

- **Bug Fixes**
  - Updated import paths for consistency without affecting functionality.

These updates enhance SSL management capabilities and improve user experience when handling certificates.

### CODE REVIEW

**Review by pshenmic — APPROVED** [review]

File: packages/dashmate/src/commands/ssl/cleanup.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        renderer: isVerbose ? 'verbose' : 'default',
+        rendererOptions: {
+          showTimer: isVerbose,
+          clearOutput: false,
+          collapse: false,
+          showSubtasks: true,
+          removeEmptyLines: false,
+        },
+      },
+    );
```
[the remark this answers is not in the document]
Comment (shumkov) [reply packages/dashmate/src/commands/ssl/cleanup.js:50]: We have progress in nested `Lister` from `cleanupZeroSSLCertificatesTask`

File: packages/dashmate/src/listr/tasks/ssl/zerossl/cleanupZeroSSLCertificatesTaskFactory.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+ * @param {listCertificates} listCertificates
+ * @param {cancelCertificate} cancelCertificate
+ * @return {cleanupZeroSSLCertificatesTask}
+ */
+export default function cleanupZeroSSLCertificatesTaskFactory(
+  listCertificates,
+  cancelCertificate,
+) {
+  /**
+   * @typedef {obtainZeroSSLCertificateTask}
```
[the remark this answers is not in the document]
Comment (shumkov) [reply packages/dashmate/src/listr/tasks/ssl/zerossl/cleanupZeroSSLCertificatesTaskFactory.js]: done

File: packages/dashmate/src/listr/tasks/ssl/zerossl/cleanupZeroSSLCertificatesTaskFactory.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+              observer.next(chalk`{green ${ctx.canceled}} / {red ${ctx.errored}} / ${ctx.total}`);
+
+              await wait(100);
+            }
+
+            if (ctx.errored > 0) {
+              observer.error(new Error('Some certificates were not canceled. Please try again.'));
+            } else {
+              observer.complete();
+            }
```
[the remark this answers is not in the document]
Comment (shumkov) [reply packages/dashmate/src/listr/tasks/ssl/zerossl/cleanupZeroSSLCertificatesTaskFactory.js:81]: this is the last task and we want to abort lister

File: packages/dashmate/src/listr/tasks/ssl/zerossl/cleanupZeroSSLCertificatesTaskFactory.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+                await cancelCertificate(ctx.apiKey, certificate.id);
+
+                ctx.canceled += 1;
+              } catch (e) {
+                ctx.errored += 1;
+
+                if (process.env.DEBUG) {
+                  // eslint-disable-next-line no-console
+                  console.warn(e);
+                }
```
[the remark this answers is not in the document]
Comment (shumkov) [reply packages/dashmate/src/listr/tasks/ssl/zerossl/cleanupZeroSSLCertificatesTaskFactory.js:69]: In case of debug we want console warning

**Review by pshenmic — APPROVED** [review]

File: packages/dashmate/src/listr/tasks/ssl/zerossl/cleanupZeroSSLCertificatesTaskFactory.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+            page += 1;
+
+            // eslint-disable-next-line no-param-reassign
+            task.output = `Found ${ctx.certificates.length} certificates`;
+          } while (certificatesPerRequest.length === 1000);
+
+          ctx.total = ctx.certificates.length;
+        },
+      },
```
[the remark this answers is not in the document]
Comment (shumkov) [reply packages/dashmate/src/listr/tasks/ssl/zerossl/cleanupZeroSSLCertificatesTaskFactory.js:48]: If it won't work we want to fail entire command

File: packages/dashmate/src/listr/tasks/ssl/zerossl/cleanupZeroSSLCertificatesTaskFactory.js [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+                if (process.env.DEBUG) {
+                  // eslint-disable-next-line no-console
+                  console.warn(e);
+                }
+              }
+
+              observer.next(chalk`{green ${ctx.canceled}} / {red ${ctx.errored}} / ${ctx.total}`);
+
+              await wait(100);
```
[the remark this answers is not in the document]
Comment (shumkov) [reply packages/dashmate/src/listr/tasks/ssl/zerossl/cleanupZeroSSLCertificatesTaskFactory.js:74]: Thank you for suggestion, but we want to add a small delay between all requests to do not overload ZeroSSL

**Pull request merged**